astropy/units/quantity.py

```diff
@@ -516,7 +525,7 @@ def __new__(cls, value, unit=None, dtype=np.inexact, copy=True, order=None,
 
         # by default, cast any integer, boolean, etc., to float
         if float_default and value.dtype.kind in 'iuO':
-            value = value.astype(float)
+            value = value.astype(float, copy=False)
```

Comment:
I don't think this changes anything because there's an incompatible dtype change...

Reply:
☑